core/src/trezor/utils.py

```diff
@@ -138,3 +138,17 @@ def obj_repr(o: object) -> str:
     else:
         d = o.__dict__
     return "<%s: %s>" % (o.__class__.__name__, d)
+
+
+def truncate_utf8(string: str, max_bytes: int) -> str:
+    """Truncate the codepoints of a string so that its UTF-8 encoding is at most `max_bytes` in length."""
+    data = string.encode()
+    if len(data) <= max_bytes:
+        return string
+
+    # Find the starting position of the last codepoint in data[0 : max_bytes + 1].
+    i = max_bytes
+    while i >= 0 and data[i] & 0xC0 == 0x80:
+        i -= 1
+
+    return data[:i].decode()
```

Comment:
Yeah, you are right. The trouble is, how would you name that function. We could also truncate to 67 bytes, although that's a bit obscure.

Reply:
Does it make sense to truncate to 128 or 256 bytes?

Reply:
We certainly can truncate to 128 or 256. I was just trying to keep it low. We can display maybe around 22 characters on one line of the Model T screen, if I recall correctly. So 64 is already way more than we can use. It would only come into play if there were multi-byte unicode characters, but we can't display those anyway. Nevertheless in that case something around 22*4=88 bytes is probably the maximum we could use. So perhaps a nice round number could be 96, 100 or 128, as you suggested.

BTW, I already added a comment explaining the value of 67: https://github.com/trezor/trezor-firmware/pull/908/commits/2bff8b11b6110e8f12c24cff8d1263e4ebf9c054.

Reply:
What about the following function:
```
def truncate_utf8(string: str, min_bytes: int) -> str:
    """Truncate the codepoints of a string so that its UTF-8 encoding is shortest possible but (if feasable) at least `min_bytes` in length."""
    data = string.encode()
    if len(data) <= min_bytes:
        return string

    # Find the ending position of the last codepoint that starts in data[0 : min_bytes].
    i = min_bytes
    while i < len(string) and data[i] & 0xC0 == 0x80:
        i += 1

    return data[:i].decode()

```

Reply:
I don't like this function very much, honestly, because it's very counterintuitive what might be the reason why you would want this. 

I am fine with 96, 100 or 128 bytes.

Reply:
I agree with @prusnak, the behavior is quite unexpected for someone who doesn't look at the function parameters and docstring carefully, making it very error prone in terms of usage.

Reply:
I changed it to 100.

Reply:
OK